pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -274,11 +280,33 @@ func (adc *attachDetachController) Run(stopCh <-chan struct{}) {
 	}
 	go adc.reconciler.Run(stopCh)
 	go adc.desiredStateOfWorldPopulator.Run(stopCh)
+
 	metrics.Register(adc.pvcLister, adc.pvLister, adc.podLister, &adc.volumePluginMgr)
+	go adc.recordADVolumesMetrics(stopCh)
```

Comment:
Please add unit tests exercising the new code in `attach_detach_controller_test.go`. Hint: this will be hard because of the way this new code is structured. Restructuring the code with interfaces/structs that implement interfaces/dependency injection patterns will allow you test it much more easily and make it more maintainable long term.